spec/jobs/sms_otp_sender_job_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+include Features::ActiveJobHelper
+
+describe SmsOtpSenderJob do
+  describe '.perform' do
+    it 'sends a message containing the OTP code to the mobile number', twilio: true do
+      TwilioService.telephony_service = FakeSms
+
+      SmsOtpSenderJob.perform_now(
+        code: '1234',
+        phone: '555-5555',
+        otp_created_at: Time.current
+      )
+
+      messages = FakeSms.messages
+
+      expect(messages.size).to eq(1)
+
+      msg = messages.first
+
+      expect(msg.from).to match(/(\+19999999999|\+12222222222)/)
+      expect(msg.to).to eq('555-5555')
+      expect(msg.body).to include('one-time passcode')
+      expect(msg.body).to include('1234')
+    end
+
+    it 'does not send if the OTP code is expired' do
+      reset_job_queues
+      TwilioService.telephony_service = FakeSms
+      FakeSms.messages = []
+      otp_expiration_period = Devise.direct_otp_valid_for
+      user = create(:user, direct_otp_sent_at: otp_expiration_period.ago)
+
+      SmsOtpSenderJob.perform_now(
+        code: '1234',
+        phone: '555-5555',
+        otp_created_at: user.direct_otp_sent_at
```

Comment:
This looks great. The only thing I would suggest is to pass in the date directly here, as opposed to having to create a user in order to pass a date. The Job doesn't know that there's a user associated with the date. It determines whether or not the date is expired by looking at the date only.